### PR TITLE
Support controlling startup via the google-logging-enable metadata key.

### PR DESCRIPTION
**Pull request by igorpeshansky**

### CODE REVIEW

**Comment by igorpeshansky** [comment]
@mr-salty Please take a look.

File: templates/etc/init.d/deb/td-agent [hunk 1]
```diff
@@ -114,9 +114,16 @@ do_start() {
   #   0 if daemon has been started
   #   1 if daemon was already running
   #   2 if daemon could not be started
+  #   3 if the daemon was not supposed to be started
   if kill_by_file -0 "${TD_AGENT_PID_FILE}"; then
     return 1
   else
+    GOOGLE_LOGGING_ENABLE=$(curl --silent -f -H "Metadata-Flavor: Google" http://169.254.169.254/computeMetadata/v1/project/attributes/google-logging-enable 2>/dev/null)
+    if [ -n "$GOOGLE_LOGGING_ENABLE" -a "$GOOGLE_LOGGING_ENABLE" -eq 0 ]; then
```
Comment (mr-salty) [inline templates/etc/init.d/deb/td-agent]: Do we need to worry about the result being non-numeric?  [ fails in this case:

```
$ if [ "foo" -eq 0 ]; then echo "ok"; fi
sh: [: foo: integer expression expected
```

File: templates/etc/init.d/rpm/td-agent [hunk 2]
```diff
@@ -184,6 +184,13 @@ do_start() {
   # Set Max number of file descriptors for the safety sake
   # see http://docs.fluentd.org/en/articles/before-install
   ulimit -n 65536 1>/dev/null 2>&1 || true
+
+  GOOGLE_LOGGING_ENABLE=$(curl --silent -f -H "Metadata-Flavor: Google" http://169.254.169.254/computeMetadata/v1/project/attributes/google-logging-enable 2>/dev/null)
+  if [ -n "$GOOGLE_LOGGING_ENABLE" -a "$GOOGLE_LOGGING_ENABLE" -eq 0 ]; then
+      echo "Disabled via metadata"
+      return 0
```
Comment (mr-salty) [inline templates/etc/init.d/rpm/td-agent]: I see the return value isn't documented here as it is for debian, is 0 what we want?  Can you add a comment.

File: templates/etc/init.d/deb/td-agent (hunk 1 above)
Comment (igorpeshansky) [inline templates/etc/init.d/deb/td-agent]: Done.

File: templates/etc/init.d/rpm/td-agent (hunk 2 above)
Comment (igorpeshansky) [inline templates/etc/init.d/rpm/td-agent]: Done.

**Comment by igorpeshansky** [comment]
PTAL.

**Comment by mr-salty** [comment]
(testing CLA webhook, please disregard)

**Comment by mr-salty** [comment]
LGTM

**Pull request merged**